crates/dojo/lang/src/derive_macros/introspect/utils.rs

```diff
@@ -41,6 +41,13 @@ pub fn get_array_item_type(ty: &str) -> String {
 /// # Examples
 ///    extract_composite_inner_type("Array<(u8, u16)", "Array<", ">") returns "u8, u16"
 pub fn extract_composite_inner_type(ty: &str, prefix: &str, suffix: &str) -> String {
+    // The unit type is a special case (empty tuple). Even if this syntax is not used much anymore,
+    // if will cause an error if a variant of an enum or a field of a struct is a unit type.
+    // `A: ()` is the same as `A`.
+    if ty == "()" {
```

Comment:
For the last field of a struct/enum, if you have `field: () // comment`, `ty` will be `() // comment`, so the `if` won't work 😕 

I think, the best solution is to replace the regex by: `{}(.*){}` as I already removed all remaining spaces with `caps[1].to_string().replace(" ", "")`.
And, with this regex, the unit type is taken into account.

Reply:
Good catch, will pass in draft to ensure we cover all the possible edge cases with tests. :thus far